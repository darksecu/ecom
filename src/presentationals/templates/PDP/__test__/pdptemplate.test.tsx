import * as React from 'react';
import {View} from 'react-native';
import {PDPTemplate} from '..';
import renderer from 'react-test-renderer';
import {fireEvent, render} from 'react-native-testing-library';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

const dummyData = {
  id: 1,
  colour: 'Black',
  name: 'Black Sheet Strappy Textured Glitter Bodycon Dress',
  price: 10,
  img: 'http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024',
};

describe('PDP Template component', () => {
  it('SnapShot', async () => {
    const expected = renderer.create(<PDPTemplate {...dummyData} />).toJSON();
    expect(expected).toMatchSnapshot();
  });

  it('Remove button is disabled by default', async () => {
    const {findByTestId} = render(<PDPTemplate {...dummyData} />);
    const removeButton = await findByTestId('removeButton');
    expect(removeButton.props.accessibilityState.disabled).toBeTruthy();
  });

  it('Quantity is 0 by default', async () => {
    const {findByTestId} = render(<PDPTemplate {...dummyData} />);
    const Quantity = await findByTestId('quanityDisplay');
    expect(Quantity.children).toEqual(['0']);
  });

  it('Price Label Validation', async () => {
    const {findByTestId} = render(<PDPTemplate {...dummyData} />);
    const data = await findByTestId('PDPPrice');
    expect(data.children).toEqual([`USD `, String(dummyData.price)]);
  });

  it('Name Label Validation', async () => {
    const {findByTestId} = render(<PDPTemplate {...dummyData} />);
    const data = await findByTestId('PDPName');
    expect(data.children).toEqual([`${dummyData.name}`]);
  });
});
