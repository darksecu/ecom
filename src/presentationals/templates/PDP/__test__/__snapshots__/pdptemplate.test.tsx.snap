// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDP Template component SnapShot 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
      }
    }
    style={
      {
        "aspectRatio": 0.5,
        "height": undefined,
        "width": 200,
      }
    }
  />
  <Text
    style={
      {
        "marginTop": 20,
      }
    }
    testID="PDPPrice"
  >
    USD 
    10
  </Text>
  <Text
    style={
      {
        "fontWeight": "600",
        "marginBottom": 20,
      }
    }
    testID="PDPName"
  >
    Black Sheet Strappy Textured Glitter Bodycon Dress
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "grey",
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="removeButton"
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        -
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "white",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        testID="quanityDisplay"
      >
        0
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "blue",
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="addButton"
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
